doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+==============================================
+Storage devices and OSD’s management workflows
+==============================================
+The cluster storage devices are the physical storage devices installed in each of the cluster’s hosts. We need to execute different operations over them and also to retrieve information about physical features and working behaviour.
+The basic use cases we have in this area are:
+
+- `1. Retrieve device information. Inventory`_
+- `2. Add OSDs`_
+- `3. Remove OSDs`_
+
+1. Retrieve device information. Inventory
+=========================================
+We must be able to review what is the current state and condition of the cluster storage devices. We need the identification and features detail(including ident/fault led on/off capable) and if the device is used or not as an OSD/DB/WAL device.
+
+The information required for each device will be at least:
+::
+
+    Hostname   Path      Type  Serial    Size   Health  Ident  Fault  Available
+
+.. Note: A more optional extended view with other information fields could be also useful.
+
+In order to know what is the current condition of the device, we need to know what is the amount of storage used, the percentage of free space, the average number of iops and  the fault led state. Part of the operational information for each device has its source in the data gathered by Prometheus. Given that the monitoring stack has reached a very important role in the Ceph Dashboard, it seems sensible to start to think in the monitoring stack as a fundamental piece of the UI installed by default.  Once we have Prometheus as an “always present” source of information, it seems sensible to start to use part of the information about performance and operational state in all the places where it can be useful to the final user.
```

Comment:
I'd argue against using prometheus here though, as this introduces weird interdependencies. We have libstoragemgmt integration in ceph-volume now. This can also provide some health metrics.

Reply:
We have several ways to obtain the information, we must use the one that provides the right information and in the more efficient way.

The important thing here is: 
"what is the operational data that can be useful to have directly in front when I examine a storage device or an OSD?"
When I talk about operational data , is the data that can help me to understand what is the behavior of the device.
For example: IOPS in an storage device
We can obtain a instant value of this attribute, but the important thing is to give the final user a view to understand easily how is working a storage device. In this case this is probably the mean, max, min, values and even the distribution of this value in the last n hours.
This is why i say that probably we need to start to deem the monitoring stack as fundamental part of a Ceph deployment.

One of the targets of this pull request is to define the list of attributes (devices/OSDs) we need have and to present to provide the best user experience.
To have all the attributes is good, but to present to the final user a list of 100 attributes is not our target ( we also should provide this bulk view ). Our target must  be to present the minimum set of attributes to understand easily what is happening and what is going to happen.

Reply:
Removed this paragraph from the document. Replaced by a paragraph pointing that all operational datos hould come from the orchestrator.